### DISCUSSION TITLE
Vstsls vs Tsserver for Lazyvim plugin ecosystem

### DISCUSSION THREAD

**Discussion by Demianeen** [LazyVim/LazyVim#2155] [General]
I recently found a few mentions of new LSP Vstsls, which as people say is faster than Tsserver. And I started to wonder why Lazyvim then used Tsserver. Are there any advantages or features that Vstsls don't have?

**Comment by dpetka2001** [comment]
Probably cuz it's a newer project (around 1 year old from what I can see on its github repo) and `tsserver` has been around longer (and possibly used by more people at the time) when LazyVim first got out. You're free to choose whatever you want in your personal configuration. Eventually, everyone should become familiar with how things are set up and make their own choices.

**Comment by Demianeen** [comment]
Perfect, understood @dpetka2001! Is it okay if I will set up tsvsls for myself and after will create typescript-tsvsls extra? Or do you think there is no need for this?

**Reply by dpetka2001** [reply]
Please do whatever you see fit for your needs. I don't make any decisions for LazyVim. I'm just a LazyVim user like you. If you have a working LazyExtra with `Vtsls`, please feel free to submit a PR if you think it'll be useful and the maintainer will decide if it will be included as an extra or not.
